src/script/components/collection/Collection.tsx

```diff
@@ -0,0 +1,225 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React, {useEffect, useState} from 'react';
+import {amplify} from 'amplify';
+import {WebAppEvents} from '@wireapp/webapp-events';
+import {registerStaticReactComponent, useKoSubscribableChildren} from 'Util/ComponentUtil';
+import {Conversation} from '../../entity/Conversation';
+
+import {ContentMessage} from 'src/script/entity/message/ContentMessage';
+import {t} from 'Util/LocalizerUtil';
+import {Category, CollectionItem, isOfCategory} from './CollectionItem';
+import Icon from 'Components/Icon';
+import FullSearch from '../../page/collection/FullSearch';
+import {ConversationRepository} from 'src/script/conversation/ConversationRepository';
+import {MessageCategory} from '../../message/MessageCategory';
+import CollectionDetails from './CollectionDetails';
+
+interface CollectionDetailsProps {
+  conversation: Conversation;
+  conversationRepository: ConversationRepository;
+}
+
+const Section: React.FC<{
```

Comment:
maybe we split this component off to a separate file also, its getting a little long in here!

Reply:
That makes sense :ok_hand: